extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/K8sDruidNodeDiscoveryProvider.java

```diff
@@ -270,9 +270,17 @@ private void keepWatching(String namespace, String labelSelector, String resourc
               if (item != null && item.type != null && item.object != null) {
                 switch (item.type) {
                   case WatchResult.ADDED:
+                    if (item.object.getNode().getServices() == null || item.object.getNode().getServices().isEmpty()) {
+                      LOGGER.warn("no services found on [%s] druid [%s] node [%s], skipping", item.type, item.object.getNode().getNodeRole(), item.object.getNode().getDruidNode().getHostAndPort());
+                      break;
+                    }
                     baseNodeRoleWatcher.childAdded(item.object.getNode());
                     break;
                   case WatchResult.DELETED:
+                    if (item.object.getNode().getServices() == null || item.object.getNode().getServices().isEmpty()) {
+                      LOGGER.warn("no services found on [%s] druid [%s] node [%s], skipping", item.type, item.object.getNode().getNodeRole(), item.object.getNode().getDruidNode().getHostAndPort());
+                      break;
+                    }
                     baseNodeRoleWatcher.childRemoved(item.object.getNode());
                     break;
```

Comment:
```suggestion
                    } else {
                      baseNodeRoleWatcher.childRemoved(item.object.getNode());
                    }
                    break;
```

Reply:
We can't change the formatting, it is inherited from Druid's formatting rules and the upstream PR will be rejected.

Reply:
They say to use `break;` rather than an `else` block?

Reply:
is this a formatting change? I just thought it was making it so there was only one `break` per switch branch...